loadbalancer/algorithm.go

```diff
@@ -145,13 +145,13 @@ func withFadeIn(rnd *rand.Rand, ctx *routing.LBContext, notFadingIndexes []int,
 type roundRobin struct {
 	mx               sync.Mutex
```

Comment:
Is mx still needed?

Reply:
Ah, ok. I guess we could also use evergrowing https://pkg.go.dev/sync/atomic#Int64 and get rid of the mutex similar to what we do here https://github.com/zalando/skipper/blob/e88336207452706b8fe92e3086a6b93b327dfbd5/ratelimit/ratelimit.go#L271-L275